sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ClassAndMethodName.cs

```diff
@@ -0,0 +1,341 @@
+                if (!char.IsLetter(suggestion[i - 1]) &&
+                    char.IsLower(suggestion[i]))
+                {
+                    suggestion[i] = char.ToUpperInvariant(suggestion[i]);
+                }
+            }
+            return suggestion.ToString();
+        }
+
+        private static string SuggestLessUppercase(string input, int maxUppercaseCount)
+        {
+            var upper = input.Take(maxUppercaseCount);
+            var lower = input.Skip(maxUppercaseCount).Select(char.ToLowerInvariant);
+
+            return new string(upper.Concat(lower).ToArray());
+        }
+
+        private static IEnumerable<string> SplitToCamelCase(string name)
+        {
+            var currentWord = new StringBuilder();
+            foreach (var c in name)
+            {
+                if (char.IsUpper(c))
+                {
+                    if (currentWord.Length > 0 && !char.IsUpper(currentWord[currentWord.Length - 1]))
+                    {
+                        yield return currentWord.ToString();
+                        currentWord.Clear();
+                    }
+
+                    currentWord.Append(c);
+                }
+                else if (char.IsLower(c))
+                {
+                    if (currentWord.Length > 1 && char.IsUpper(currentWord[currentWord.Length - 1]))
+                    {
+                        var lastChar = currentWord[currentWord.Length - 1];
+                        currentWord.Length--;
+                        yield return currentWord.ToString();
+                        currentWord.Clear();
+                        currentWord.Append(lastChar);
+                    }
+
+                    currentWord.Append(c);
+                }
+                else
+                {
+                    if (currentWord.Length > 0)
+                    {
+                        yield return currentWord.ToString();
+                        currentWord.Clear();
+                    }
+
+                    yield return c.ToString();
+                    continue;
+                }
+            }
+
+            if (currentWord.Length > 0)
+            {
+                yield return currentWord.ToString();
+            }
+        }
+
+        private static bool IsTypeComRelated(INamedTypeSymbol symbol)
+        {
+            return symbol == null ||
+                symbol.GetAttributes().Any(a =>
+                    a.AttributeClass.Is(KnownType.System_Runtime_InteropServices_ComImportAttribute) ||
+                    a.AttributeClass.Is(KnownType.System_Runtime_InteropServices_InterfaceTypeAttribute));
+        }
+
+        private static string ToInitCap(string input)
+        {
+            return input.Length > 0
+                ? char.ToUpperInvariant(input[0]) + input.Substring(1)
+                : input;
+        }
+
+        private static bool IsCharUpper(string input, int idx)
+        {
+            return idx < 0 || idx >= input.Length
+                ? false
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the unnecessary Boolean literal(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1125)